File: LeetCode/Medium/3871-count-commas-in-range-ii/3871-count-commas-in-range-ii.py
class Solution:
    def countCommas(self, n: int) -> int:
        commas_count = 0
        lower = 1000
        commas_in_number = 1

        while lower <= n:
            commas_count += (min(n, lower * 1000 - 1) - lower + 1) * commas_in_number
            lower *= 1000
            commas_in_number += 1



        # commas_count = 0
        # threshold = 1000

        # while threshold <= n:
        #     commas_count += n - threshold + 1
        #     threshold *= 1000
        
        return commas_count

        

File: LeetCode/Medium/3871-count-commas-in-range-ii/3871-count-commas-in-range-ii.cs
public class Solution {
    public long CountCommas(long n) {
        long commasCount = 0;
        long threshold = 1000;

        while (threshold <= n)
        {
            commasCount += n - threshold + 1;
            threshold *= 1000;
        }
        
        return commasCount;
    }
}